src/evaluate_fences.rs

```diff
@@ -100,6 +98,13 @@ pub fn evaluate_fences<'fencecollectionlifetime, 'sourcefilelifetime>(
                 // fences of the file we are importing.
                 ResolvedImport::ProjectLocalImport(project_local_path) => {
                     let project_local_path_str = project_local_path.to_str().unwrap();
+                    if ignored_dirs
+                        .iter()
+                        .any(|d| d.is_match(project_local_path_str))
+                    {
+                        continue;
```

Comment:
This should all be redundant now,  fence evaluation does not need to be aware of ignored files or fences, since they won't be picked up during traversal

Reply:
I think this is either needed or we need a better way to handle the following case:
if `ignoredDirs = ['shared/...']` and `project_local_path_str ="shared/foo/bar/index.ts"` this will always throw an error. Because the import is resolved to non `node_modules` path but the file isn't in the `source_files_map`

Reply:
I would say we should error when importing ignored code. The code is meant to be totally ignored, and we can't provide safety guarantees on anything that we're ignoring.

Reply:
After looking into this, we are already throwing an error when a resolved import is not found in the `source_file_map`
```
could not find project local path shared/internal/owa-client-types/src/index imported by packages/framework/owa-connected-account-headers/src/utils/getConnectedAccountWebSessionType.ts with specifier owa-client-types
```
Maybe having an error enum that differentiates `fence violation error` from `fence evaluation error`?